homeassistant/components/stream/__init__.py

```diff
@@ -215,6 +215,9 @@ def stop(self):
         if self._hls:
             self._hls.cleanup()
             self._hls = None
+        if self._hls_timer:
```

Comment:
So if we move this above line 215 we could take out the change to line 143?

Reply:
Nevermind, we would have to wait for the call later

Reply:
Yes, good point, if we clear the timer on stop then the idle callback should only ever be invoked with an active hls stream. I've removed the check above and moved this up before cleanup as you've suggested.  I like this as it makes the idle callback contract more clear / explicit.

Reply:
You had another comment that said nevermind and we have to wait for the call -- is there a race here?

Reply:
No sorry I misread it..I think this is good